### PR TITLE
Add Supertokens migration page.

### PR DESCRIPTION
**Pull request by ConzorKingKong**
## Summary of change
Add supertokens core and SDK migration links to github changelogs

## Checklist
- [ ] Checked for broken links? (Run `cd v2 && MODE=production npx docusaurus build`)

### CODE REVIEW

File: v2/community/migration/migrating-supertokens.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+- ### Python
+[supertokens-python](https://github.com/supertokens/supertokens-python/blob/master/CHANGELOG.md)
+
+## Front-end SDK Migration
+
+- ### ReactJS
+[supertokens-auth-react](https://github.com/supertokens/supertokens-auth-react/blob/master/CHANGELOG.md)
+
+- ### Website
+[supertokens-website](https://github.com/supertokens/supertokens-website/blob/master/CHANGELOG.md)
```
Comment (rishabhpoddar) [inline v2/community/migration/migrating-supertokens.mdx:36]:
```suggestion
```

File: v2/community/migration/migrating-supertokens.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
+id: migrating-supertokens
+title: Migrating Supertokens
```
Comment (rishabhpoddar) [inline v2/community/migration/migrating-supertokens.mdx:3]:
```suggestion
title: Updating Supertokens SDKs and Core
```

File: v2/community/migration/migrating-supertokens.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+---
+id: migrating-supertokens
```
Comment (rishabhpoddar) [inline v2/community/migration/migrating-supertokens.mdx:2]:
```suggestion
id: updating-supertokens
```

File: v2/community/migration/migrating-supertokens.mdx [hunk 4]
```diff
@@ -0,0 +1,51 @@
+---
```
Comment (rishabhpoddar) [inline v2/community/migration/migrating-supertokens.mdx:1]: this page needs to be there in each of the recipe guides as well - as a separate section, between the existing Migration and References section. Also, rename the existing Migration section to "Migrating to SuperTokens"

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: v2/community/updating-supertokens.mdx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+Not all older releases of every supertokens Core, front-end SDK, and back-end SDK are compatible with each other. In order to maintain compatability
+and keep up on new features, we recommend updating and upgrading all of your SDKs regularly, starting with the supertokens core, then the back-end SDK, then the front-end SDK.
+
+The following is a link to each core and SDK changelog, containing steps to migrate from each version. You can use our [compatability table](../../community/compatibility-table) to see which versions you need to upgrade to.
+
+## Core Migration
+
+- ### Core
+[supertokens-core](https://github.com/supertokens/supertokens-core/blob/master/CHANGELOG.md)
```
Comment (rishabhpoddar) [inline v2/community/updating-supertokens.mdx:20]: you should say if using managed service, please email us on team@supertokens.com from your registered email ID requesting a core update. For seld hosted, see the changelog

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: v2/phonepassword/sidebars.js [hunk 6]
```diff
@@ -12,6 +12,7 @@ module.exports = {
         "backend/passwordless-customisation"
       ],
     },
-    "frontend"
+    "frontend",
+    "updating-supertokens"
```
Comment (rishabhpoddar) [inline v2/phonepassword/sidebars.js:16]: can remove it from this recipe.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: v2/passwordless/updating-supertokens.mdx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+<!-- COPY DOCS -->
+<!-- ./community/updating-supertokens.mdx -->
+
+# Updating Supertokens SDKs and Core
+
+Not all older releases of every supertokens Core, front-end SDK, and back-end SDK are compatible with each other. In order to maintain compatability
+and keep up on new features, we recommend updating and upgrading all of your SDKs regularly, starting with the supertokens core, then the back-end SDK, then the front-end SDK.
+
+The following is a link to each core and SDK changelog, containing steps to migrate from each version. You can use our [compatability table](../../community/compatibility-table) to see which versions you need to upgrade to.
```
Comment (rishabhpoddar) [inline v2/passwordless/updating-supertokens.mdx:15]: if you visit this link on any of the recipe guides, its broken..

File: v2/passwordless/updating-supertokens.mdx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+- ### Self-hosted Core
+[supertokens-core](https://github.com/supertokens/supertokens-core/blob/master/CHANGELOG.md)
+
+- ### Managed Core
+If you're using the managed core service, please email us at team@supertokens.com from your registered email to request a core update.
+
+## Back-end SDK Migration
+
+- ### NodeJS
+[supertokens-node](https://github.com/supertokens/supertokens-node/blob/master/CHANGELOG.md)
```
Comment (rishabhpoddar) [inline v2/passwordless/updating-supertokens.mdx:28]: please dont just link to the changelog.. you need to tell users that this is the changelog and to look for the migration section under each version from the one they have uptil the one they want to move to.

File: v2/passwordless/updating-supertokens.mdx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+The following is a link to each core and SDK changelog, containing steps to migrate from each version. You can use our [compatability table](../../community/compatibility-table) to see which versions you need to upgrade to.
+
+## Core Migration
+
+- ### Self-hosted Core
+[supertokens-core](https://github.com/supertokens/supertokens-core/blob/master/CHANGELOG.md)
+
+- ### Managed Core
+If you're using the managed core service, please email us at team@supertokens.com from your registered email to request a core update.
```
Comment (rishabhpoddar) [inline v2/passwordless/updating-supertokens.mdx:23]: This needs more clarity.. talk to @sattvikc here about how to actually run the migrations - suppose that users want to migrate from v5 to v9, do the users need to stop the core, run all the migrations from v5...v9 that are mentioned in the changelog, and then start the v9 core? Or stop v5, do migration of next version, start core of next version, then do migration of version after that, then start core for that version.. all the way till v9?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: v2/passwordless/updating-supertokens.mdx (hunk 7 above)
Comment (ConzorKingKong) [reply v2/passwordless/updating-supertokens.mdx:15]: I had hardcoded the link to the updating-supertokens page before and forgot to update this after I set up page copying, good catch. I'll fix this

**Comment by ConzorKingKong** [comment]
For the session addon - Since the upgrade guide is linked in the SDK reference, I have to put “upgrading supertokens” in the sidebar or else the sidebar disappears when clicking the link to it from the sdk reference page

File: v2/passwordless/updating-supertokens.mdx (hunk 9 above)
Comment (ConzorKingKong) [reply v2/passwordless/updating-supertokens.mdx]: Talked with him and updated the copy

File: v2/community/upgrading-supertokens.mdx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+:::
+
+In order to maintain compatibility and keep up on new features, we recommend upgrading your core and SDKs whenever possible.
+
+## 1. Upgrade your core
+
+You will need to shut down your core in order to upgrade it. After it's off, run the migration scripts for all major versions inbetween
+your starting version and the version you are upgrading to. If the supertokens core version you have upgraded to shows as compatible
+with your current front-end and back-end sdk's in the [compatibility table](./compatibility-table), then you can bring your core back online.
+Otherwise, please leave your core offline as you continue to upgrade your other SDK's.
```
Comment (rishabhpoddar) [inline v2/community/upgrading-supertokens.mdx:23]: - doesnt make sense to tell people to leave the core offline if their backend sdk is not compatible with the core.. you need to instead tell them to keep the core online and then not start the backend until the backend sdk is upgraded.
- There are migration scripts in minor update versions as well (See changelog..)

File: v2/community/upgrading-supertokens.mdx [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+- ### Self-hosted core
+[supertokens-core](https://github.com/supertokens/supertokens-core/blob/master/CHANGELOG.md)
```
Comment (rishabhpoddar) [inline v2/community/upgrading-supertokens.mdx:26]:
```suggestion
[See supertokens-core changelog](https://github.com/supertokens/supertokens-core/blob/master/CHANGELOG.md)
```

Comment (rishabhpoddar) [reply v2/community/upgrading-supertokens.mdx:26]: similar changes to all the links below

**Comment by rishabhpoddar** [comment]
> For the session addon - Since the upgrade guide is linked in the SDK reference, I have to put “upgrading supertokens” in the sidebar or else the sidebar disappears when clicking the link to it from the sdk reference page

Huh? Thats true for all the recipes you have modified, correct?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Comment by ConzorKingKong** [comment]
> > For the session addon - Since the upgrade guide is linked in the SDK reference, I have to put “upgrading supertokens” in the sidebar or else the sidebar disappears when clicking the link to it from the sdk reference page
> 
> Huh? Thats true for all the recipes you have modified, correct?

yes, but you had asked me to only put it in the sidebar for auth recipes and sessions is an addon recipe. so just clarifying that I had to do it for an addon recipe as well

File: v2/community/upgrading-supertokens.mdx (hunk 11 above)
Comment (ConzorKingKong) [reply v2/community/upgrading-supertokens.mdx]: done

File: v2/community/upgrading-supertokens.mdx (hunk 10 above)
Comment (ConzorKingKong) [reply v2/community/upgrading-supertokens.mdx]: That is much more sensible. I updated it

**Pull request merged**